libqpdf/QPDFObjectHandle.cc

```diff
@@ -3115,27 +3111,13 @@ QPDFObjectHandle::assertPageObject()
 bool
 QPDFObjectHandle::dereference()
 {
-    if (!this->initialized) {
+    if (!initialized) {
         return false;
     }
-    if (this->obj.get() && getObjectID() &&
-        QPDF::Resolver::objectChanged(this->qpdf, getObjGen(), this->obj)) {
-        this->obj = nullptr;
-    }
-    if (this->obj.get() == nullptr) {
-        std::shared_ptr<QPDFObject> obj =
-            QPDF::Resolver::resolve(this->qpdf, getObjGen());
-        if (obj.get() == nullptr) {
-            // QPDF::resolve never returns an uninitialized object, but
-            // check just in case.
-            this->obj = QPDF_Null::create();
-        } else if (dynamic_cast<QPDF_Reserved*>(obj.get())) {
-            // Do not resolve
```

Comment:
By removing this, you are causing the code to have to hang unto a bunch of reserved objects that are going to get replaced later. It's possible that the original code was made functionally unnecessary by the checks for whether the object was changed (a relatively recent addition), but I wonder if there is a performance impact. It's possible that the old code is superfluous.